### PR TITLE
General Parser implementation

### PR DESCRIPTION
**Pull request by suflors**
Adds a general SentenceParser class that can parse plain text into tokenized sentences for optimum markov efficiency.

### CODE REVIEW

**Comment by Iron-Squid** [comment]
wow this looks like code

**Comment by tcheerno** [comment]
That is certainly written in .net yes

**Comment by aliceif** [comment]
If it has the concept of escaped characters it's not plain text is it? This needs some rethinking.

**Comment by suflors** [comment]
> If it has the concept of escaped characters it's not plain text is it? This needs some rethinking.

Handled in recent commits, now there is no escapement.

File: AtelierTomato.Markov/AtelierTomato.Markov.Parser/SentenceParser.cs [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+(?<=\s[!?]*)[!?][!?]+    # questionexciteclusters
+|                        # nor
+[.!?]\w                  # punctuation with word after it
+)+)                      # 1 or multiple
+(\.|[.!?]+|$|\r?\n)", RegexOptions.IgnorePatternWhitespace | RegexOptions.Compiled);
+		private readonly Regex spaceifyEllipsesPattern = new Regex(@"(?<=[^\s.,?!])([.,?!])(?=[.,?!])", RegexOptions.Compiled);
+		private readonly Regex ignoreCountPattern = new Regex(@"^[\p{P}]*$", RegexOptions.Compiled);
+		private readonly Regex deleteLinkPattern = new Regex(@"\S*://\S*", RegexOptions.Compiled);
+		private readonly Regex whitespaceCleanerPattern = new Regex(@"[^\S\r\n]+", RegexOptions.Compiled);
+		private readonly Regex detahQuoteArrowsPattern = new Regex(@"(?<=^|\n)(>)(?=\S)", RegexOptions.Compiled);
```
Comment (aliceif) [inline AtelierTomato.Markov/AtelierTomato.Markov.Parser/SentenceParser.cs]: we typo'd "detach" there

**Review by aliceif — APPROVED** [review]

**Pull request merged**